pkg/reconciler/events/cloudauditlog/cloudauditlog.go

```diff
@@ -0,0 +1,274 @@
+		return nil
+	}
+
+	// Get the CloudAuditLog resource with this namespace/name
+	original, err := c.cloudauditlogLister.CloudAuditLogs(namespace).Get(name)
+	if apierrs.IsNotFound(err) {
+		// The CloudAuditLog resource may no longer exist, in which case we stop processing.
+		runtime.HandleError(fmt.Errorf("CloudAuditLog '%s' in work queue no longer exists", key))
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	// Don't modify the informers copy
+	cal := original.DeepCopy()
+
+	reconcileErr := c.reconcile(ctx, cal)
+
+	// If no error is returned, mark the observed generation.
+	if reconcileErr == nil {
+		cal.Status.ObservedGeneration = cal.Generation
+	}
+
+	// If we didn't change anything then don't call updateStatus.
+	// This is important because the copy we loaded from the
+	// informer's cache may be stale and we don't want to
+	// overwrite a prior update to status with this stale state.
+	if !equality.Semantic.DeepEqual(original.Status, cal.Status) {
+		if _, err := c.updateStatus(ctx, cal); err != nil {
+			c.Logger.Warn("Failed to update CloudAuditLog Source status", zap.Error(err))
+			c.Recorder.Eventf(cal, corev1.EventTypeWarning, "UpdateFailed",
+				"Failed to update status for CloudAuditLog %q: %v", cal.Name, err)
+			return err
+		} else if reconcileErr == nil {
+			c.Recorder.Eventf(cal, corev1.EventTypeNormal, "Updated", "Updated CloudAuditLog %q", cal.Name)
+		}
+	}
+
+	if reconcileErr != nil {
+		c.Recorder.Event(cal, corev1.EventTypeWarning, "InternalError", reconcileErr.Error())
+	}
+	return reconcileErr
+}
+
+func (c *Reconciler) reconcile(ctx context.Context, cal *v1alpha1.CloudAuditLog) error {
+	topic := cal.Status.TopicID
+	if topic == "" {
+		topic = fmt.Sprintf("cloudauditlog-%s", string(cal.UID))
+	}
+	cal.Status.InitializeConditions()
+
+	// See if the source has been deleted.
+	if cal.DeletionTimestamp != nil {
+		err := c.deleteSink(ctx, cal)
+		if err != nil {
+			cal.Status.MarkSinkNotReady("SinkDeleteFailed", "Failed to delete Stackdriver sink: %s", err.Error())
+			return fmt.Errorf("failed to delete Stackdriver sink: %v", err)
+		}
+		cal.Status.MarkSinkNotReady("SinkDeleted", "Successfully deleted Stackdriver sink: %s", cal.Status.SinkID)
+		cal.Status.SinkID = ""
+
+		err = c.PubSubBase.DeletePubSub(ctx, cal)
+		if err != nil {
+			return fmt.Errorf("failed to delete pubsub resources: %s", err)
+		}
+		c.removeFinalizer(cal)
+		return nil
+	}
+
+	// Ensure the finalizer's there, since we're about to attempt
+	// to change external state with the topic, so we need to
+	// clean it up.
+	c.ensureFinalizer(cal)
+	t, ps, err := c.PubSubBase.ReconcilePubSub(ctx, cal, topic, resourceGroup)
+	if err != nil {
+		return err
+	}
+	c.Logger.Infof("Reconciled: PubSub: %+v PullSubscription: %+v", t, ps)
+
+	sink, err := c.reconcileSink(ctx, cal)
+	if err != nil {
+		return err
+	}
+	cal.Status.SinkID = sink
+	cal.Status.MarkSinkReady()
+	c.Logger.Infof("Reconciled Stackdriver sink: %+v", sink)
```

Comment:
remove

Reply:
Do you think I should remove other similar info logs from the reconciler?

Reply:
I'll answer since @nachocano is out: I don't know which other similar info logs there are, but this particular log should be debug level.

Reply:
Done.